scraper_service/main.py

```diff
@@ -1,79 +1,90 @@
+import json
+import logging
 from datetime import datetime
 from time import sleep
+
+from pytz import timezone
 from botocore.vendored import requests
-from fake_useragent import UserAgent
 import boto3
-import json
+from fake_useragent import UserAgent
 
+URL = 'http://cdn.shakeshack.com/camera.jpg'
+PREDICT_TOPIC_ARN = 'arn:aws:sns:us-east-1:245636212397:triggerPredict'
+SMS_TOPIC_ARN = 'arn:aws:sns:us-east-1:245636212397:triggerSMS'
+S3_BUCKET = 'deepshack'
+
+logging.getLogger().setLevel(logging.INFO)
 
-def make_request():
 
-    url = 'http://cdn.shakeshack.com/camera.jpg'
+def make_request():
+    """Sends GET request to the ShackCam and returns response object"""
     headers = {'user-agent': UserAgent().random}
-    r = requests.get(url, stream=True, headers=headers)
+    r = requests.get(URL, stream=True, headers=headers)
     if r.headers['content-length'] == '0':
         raise ValueError('No image was loaded, trying again...')
     else:
         return r
 
 
-def scrape_handler(event, context):
+def publish_message(message, topic_arn):
+    """Publishes a message to SNS topic"""
+    sns = boto3.client('sns')
+    response = sns.publish(
+        TopicArn=topic_arn,
+        Message=json.dumps({'default': json.dumps(message)}),
+        MessageStructure='json'
+    )
+    logging.info(response)
 
-    # Receive message from inbound_SMS_service on the lexMessage SNS topic:
-    event_message = json.loads(event['Records'][0]['Sns']['Message'])
-    # event_message = event['Records'][0]['Sns']['Message']
 
-    # Get phone number:
-    phone_raw = event_message['from']
-    phone = phone_raw.split("B")[-1]
+def scrape_handler(event, context):
+    """
+    Main function that gets called by Lambda. Subscribes to a message topic,
+    triggerPredict, scrapes an image, saves to AWS S3, and then publishes
+    messages to triggerPredict and triggerSMS
+    """
 
-    # Get image filename:
-    timestamp = datetime.now()
-    timestamp_str = timestamp.strftime('%Y-%m-%d_%H%M_')
-    filename = f'shackcam/{timestamp_str}' + phone + '.jpg'
+    # Receive message from upstream service
+    event_message = event['Records'][0]['Sns']['Message']
+    phone_number = event_message['phone_number'].split("B")[-1]
 
-    # Create message and send to predict_service_trigger
-    # on topic "triggerPredict"
-    outbound_message = {'image_id': filename,
-                        'phone': phone}
+    timestamp = datetime.now(timezone('EST'))
```

Comment:
Because we're not using pytz, replace this line with:

timestamp = datetime.now()

Reply:
All set